packages/backend-ai/src/get_message.ts

```diff
@@ -0,0 +1,115 @@
+import {
+  BedrockRuntimeClient,
+  ConverseCommand,
+  ConverseCommandInput,
+  Message,
+} from '@aws-sdk/client-bedrock-runtime';
+import {
+  ContentBlock,
+  GetMessageInput,
+  GetMessageOutput,
+  Tool,
+} from './types.js';
+
+/**
+ * Sends a message to the AI model and retrieves a response.
+ * This function leverages ConverseCommand (https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/client/bedrock-runtime/command/ConverseCommand/)
+ * @param params - The parameters for the message.
+ * @param params.systemPrompts - System prompts to guide the AI's behavior.
+ * @param params.messages - The conversation history.
+ * @param params.modelId - The ID of the AI model to use.
+ * @param [params.tools] - Optional tools that the AI can use.
+ * @returns The AI's response and associated metadata.
+ * @throws Throws an error if the AI's response does not contain a message.
+ */
+export const getMessage = async ({
+  systemPrompts,
+  messages,
+  modelId,
+  tools,
+}: GetMessageInput): Promise<GetMessageOutput> => {
+  const client = new BedrockRuntimeClient();
+
+  const input: ConverseCommandInput = {
+    modelId,
+    messages: messages.map((msg) => ({
+      role: msg.role,
+      content: msg.content.map(convertContentBlock),
+    })) as Message[],
+    system: systemPrompts,
+    ...(tools && tools.length > 0
+      ? {
+          toolConfig: {
+            tools: tools.map(convertTool),
+            toolChoice: { auto: {} },
+          },
+        }
+      : {}),
+  };
+
+  const command = new ConverseCommand(input);
+  const output = await client.send(command);
+
+  if (!output.output?.message) {
+    throw new Error('No message in ConverseCommandOutput');
```

Comment:
The messaging should probably not leak the `ConverseCommandOutput` terminology. Something more general like "The model response did not contain a message" maybe?